Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs

```diff
@@ -0,0 +1,261 @@
+        public void Should_succeed_when_asserting_statuscode_is_successful(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeSuccessful();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRedirectionStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_redirect(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeRedirection();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetClientErrorStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_client_error(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeClientError();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetServerErrorStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_server_error(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeServerError();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetClientErrorStatusCodes))]
+        [MemberData(nameof(GetServerErrorStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_error(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeError();
+        }
+
+        [Fact]
+        public void Should_fail_when_asserting_statuscode_error_is_successful()
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(HttpStatusCode.Gone);
+
+            // Act
+            Action action = () => testee.Should().BeSuccessful();
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void Should_fail_with_descriptive_message_when_asserting_statuscode_error_is_successful()
+        {
+            // Act
+            Action action = () =>
+                new HttpResponseMessage(HttpStatusCode.Gone).Should().BeSuccessful("because we want to test the failure {0}", "message");
+
+            // Assert
+            action
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected HttpStatusCode to be successful (2xx) because we want to test the failure message, but found HttpStatusCode.Gone {value: 410}.");
+        }
+
+        [Fact]
+        public void Should_fail_when_asserting_statuscode_error_is_redirection()
```

Comment:
🔧 I suggest to group the success and failure cases for the same methods together.